### PR TITLE
Track artifact configuration in loadouts

### LINKED ISSUE #9255: Snapshot relevant artifact configuration with loadouts
Looks like we can maybe find out what the unlocked artifact nodes are from character progressions. It'd be great if we could automatically track the artifact unlocks that are relevant to a load out when we save a load out.

### PR DESCRIPTION
**Pull request by robojumper**
Fixes #9255.

grafik

No interface for picking a bunch of mods, only the "Sync from equipped" button and a way to remove individual mods for now.

### CODE REVIEW

File: src/@destinyitemmanager/dim-api-types.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import '@destinyitemmanager/dim-api-types';
+// Extensions/customizations to the DIM types
+
+declare module '@destinyitemmanager/dim-api-types' {
+  export interface LoadoutParameters {
+    /** The artifact unlocks relevant to this build. */
+    artifactUnlocks?: number[];
+  }
+}
```
Comment (robojumper) [inline src/@destinyitemmanager/dim-api-types.d.ts:9]: Needs to be moved into dim-api.

**Comment by bhollis** [comment]
Incredible! I'm super into it. One thought while I'm on my phone and unable to effectively review the code - it might be nice to record the artifact's item hash or something similar here, so when a new season comes around we can more easily message what it was configured for. I have no idea whether unlocks will be reused between artifacts but it seems like something that could be valuable to have. Or maybe even just recording the season.

**Comment by robojumper** [comment]
> record the artifact's item hash

So the profile response includes an artifact definition hash, `2894222926`, which is named "Bell of Conquests" and is the only entry in the `Artifacts` table. Similarly, the vendor ("The Gate Lord's Eye"/`2894222926`) is also updated every season. An option would be getting the character's current artifact item hash (nevermind the fact that I have two [1]).

[1] grafik

**Comment by bhollis** [comment]
Ah, got it. They're not gonna make this easy are they? Perhaps just recording the current season when it's set is better then.

File: src/app/loadout/loadout-edit/LoadoutEdit.tsx [hunk 2]
```diff
@@ -82,6 +92,8 @@ export default function LoadoutEdit({
     [itemCreationContext, loadout.items, store, allItems, modsByBucket]
   );
 
+  const artifactUnlocks = profileResponse && getArtifactUnlocks(profileResponse, store.id);
```
Comment (bhollis) [inline src/app/loadout/loadout-edit/LoadoutEdit.tsx:95]: Not sure if this would be better off pushed up to a selector.

**Review by bhollis — APPROVED** [review]
Wonderful!

**Pull request merged**